app/src/main/java/com/github/se/polyfit/ui/components/scaffold/SimpleTopBar.kt

```diff
@@ -18,11 +19,25 @@ import androidx.compose.ui.platform.testTag
 import com.github.se.polyfit.ui.theme.PrimaryPurple
 import com.github.se.polyfit.ui.utils.titleCase
 
-// TODO - Replace other Simple TopBars with this, used it in AdditionalMealInfoScreen.kt
+/**
+ * A simple top bar with a back button and a title.
+ *
+ * @param title The title of the top bar.
+ * @param navigateBack The action to perform when the back button is clicked.
+ */
 @OptIn(ExperimentalMaterial3Api::class)
 @Composable
 fun SimpleTopBar(title: String, navigateBack: () -> Unit) {
   var goBackClicked by remember { mutableStateOf(false) }
+
+  // Delay the goBackClicked state change to prevent multiple back navigation.
+  LaunchedEffect(key1 = goBackClicked) {
+    if (goBackClicked) {
+      kotlinx.coroutines.delay(1000)
+      goBackClicked = false
+    }
+  }
```

Comment:
Usually it is not a great idea to use a delay in your code. Can't you achieve the same functionality using something like the previous implementation ?

Reply:
The previous implementation blocks the button from ever being pressed again, which is not good for what we want. I researched and this is the only suggested way to achieve this. Do you have anything in mind?

Reply:
I did some research, and it says that delay is bad if it blocks the main thread, since users have to wait longer than needed. But, for a UI element in a non-blocking thread (coroutine) it is ok.

"Yes, if you're using it to wait for a result, that is quite terrible. Would have to see how you're actually using it to understand for sure if there's a problem. delay() in a fragment is totally appropriate if it has to do with UI appearances." from https://stackoverflow.com/questions/76155021/is-it-a-good-practice-to-use-the-kotlin-delay-method-inside-a-fragment